core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -179,7 +181,7 @@
     @Nullable
     private Long shmSize;
 
-    private Map<MountableFile, String> copyToFileContainerPathMap = new HashMap<>();
+    private List<Entry<MountableFile, String>> copyToFileContainerPathList = new ArrayList<>();
```

Comment:
If we used a `LinkedHashMap` wouldn't we still get the benefits of deterministic ordering without having to bring `Entry` into play? The PR would just be a one-line change, wouldn't it?

Also, one tiny benefit of using a `Map` would be deduplication, although this is arguably not that big a deal.

Reply:
@rnorth thanks for looking into it.

see
https://github.com/prestosql/presto/blob/f648b877eafbfff297c235cb2af8f19ef84164ad/presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java#L35-L36

so

- yes
- but we should switch to List anyway, since it should be completely legal to copy one source file into multiple destinations in a container
  - currently, last copy wins, previous are clobbered.

Reply:
Ah, yes, re your second point we went around in circles on this in #2144. The map key/value are the wrong way round (currently) for the deduplication to be useful.

Fair enough - I'm fine with this then. Thanks for the clarification.